src/map/utils/charutils.cpp

```diff
@@ -4715,6 +4716,29 @@ namespace charutils
         }
     }
 
+    bool IsAidBlocked(CCharEntity* PInitiator, CCharEntity* PTarget) {
+        if (PTarget->getBlockingAid())
+        {
+            bool inAlliance = false;
+            PTarget->ForAlliance([&PInitiator, &PTarget, &inAlliance](CBattleEntity* PEntity) {
+                if (PEntity->id == PInitiator->id)
+                {
+                    inAlliance = true;
+                }
+            });
+
+            if (!inAlliance)
+            {
+                // Target is blocking assistance
+                PInitiator->pushPacket(new CMessageSystemPacket(0, 0, 225));
```

Comment:
let the caller push the packets - you don't know how they will need to handle it (for example, IsValidTarget using errmsg instead of pushing packets directly)